build-release-tools/application/release_npm_packages.py

```diff
@@ -0,0 +1,230 @@
+            package_dir = os.path.join(build_dir, key)
+            print "starting to update version of {0} to {1}".format(package_dir, value)
+            NPM.update_version(package_dir, version=value)
+    except Exception, e:
+        raise RuntimeError("Failed to update version for package under {0} \ndue to {1}".format(build_dir, e))
+
+def update_packages_dependency(build_dir, version_dict):
+    """
+    Update the dependency of package.json.
+    For example:
+    - "on-core": "git+https://github.com/RackHD/on-core.git",
+    + "on-core": "1.0.0",
+    """
+    try:
+        for key, value in version_dict.items():
+            package_dir = os.path.join(build_dir, key)
+            print "starting to update the dependency of {0}".format(package_dir)
+            update_dependency(package_dir, version_dict)
+    except Exception, e:
+        raise RuntimeError("Failed to update dependency for package under {0} \ndue to {1}".format(build_dir, e))
+
+
+def update_dependency(package_dir, version_dict):
+    package_json_file = os.path.join(package_dir, "package.json")
+    if not os.path.exists(package_json_file):
+        # If there's no package.json file, there is nothing more for us to do here
+        return
+
+    changes = False
+    log = ""
+    with open(package_json_file, "r") as fp:
+        package_data = json.load(fp)
+        if "dependencies" in package_data:
+            for package, version in package_data["dependencies"].items():
+                new_version = None
+                if package in version_dict:
+                    new_version = version_dict[package]
+                if new_version is not None and version != new_version:
+                    package_data["dependencies"][package] = new_version
+                    changes = True
+                    log += "  {0}:\n    WAS {1}\n    NOW {2}\n".format(package,version,new_version)
+
+    if changes:
+        print "There are changes to dependencies for {0}\n{1}".format(package_json_file, log)
+        os.remove(package_json_file)
+        new_file = package_json_file
+        with open(new_file, "w") as newfile:
+            json.dump(package_data, newfile, indent=4, sort_keys=True)
+    else:
+        print "There are NO changes to data for {0}".format(package_json_file)
+
+def publish_packages(build_dir, is_official_release, npm_registry, npm_token):
+    """
+    Publish all the npm packages under directory specified by build_dir
+    """
+    npm_packages = get_npm_packages(build_dir)
+    # Skip the release of di.js
+    npm_packages.remove("di.js")
+    for package in npm_packages:
+        try:
+            package_dir = os.path.join(build_dir, package)
+            npm = NPM(npm_registry, npm_token)
+            print "starting to publish npm package {0}".format(package)
+            if is_official_release:
+                npm.publish_package(package_dir)
+            else:
+                npm.publish_package(package_dir, tag="ci-release")
```

Comment:
The tag "ci-release" is used to install the latest version of nightly build.
If users want to install the latest stable release, they need to run ```npm install package```
If users want to install the latest nightly release, they need to run ```npm install package@ci-release```

The npm registry: https://www.npmjs.com/
You can get the details about a npm package by running ```npm view package```